server/spec/entitlement_resource_spec.rb

```diff
@@ -193,4 +193,36 @@
     end.should raise_exception(RestClient::Forbidden)
   end
 
+  it 'should handle concurrent requests to pool and maintain quanities' do
+    owner_client = user_client(@owner, random_string('owner'))
+    prod = create_product(random_string('product'), random_string('product'),
+      {:attributes => { :'multi-entitlement' => 'yes'},
+       :owner => @owner['key']})
+    pool = create_pool_and_subscription(@owner['key'], prod.id, 50)
+
+    t1 = Thread.new{register_and_consume(pool, "system", 5)}
+    sleep 1
+    t2 = Thread.new{register_and_consume(pool, "candlepin", 7)}
+    sleep 1
+    t3 = Thread.new{register_and_consume(pool, "system", 6)}
+    sleep 1
+    t4 = Thread.new{register_and_consume(pool, "candlepin", 11)}
+    t1.join
+    t2.join
+    t3.join
+    t4.join
```

Comment:
Are the sleeps necessary here? Thread.join will make the main thread wait for all the binds to complete.